### PR TITLE
Made it possible to scan spellbooks into ereader

### PR DESCRIPTION
**Pull request by Skrassh**
#### Summary
Features "Replaced the has_identified field with is_scannable for ebookscan."

#### Purpose of change
Spell books currently cant be scanned into e-readers because you have to identify a book before being able to scan it. I replaced this with is_scannable which checks for the json field scannable. Jsonizing this allows for any book marked with that to be scanned into the ereader, including spellbooks now.

#### Describe the solution
I replaced the has_identified requirement from ereader scan with is_scannable and jsonized it. I also added a check when finishing a scan to mark the book as identified if it isn't already marked.

#### Describe alternatives you've considered
Making spell books spawn unidentified so they could be identified and then scanned in, that felt like an awkward band-aid to the problem.

#### Testing
1. Create new character
2. Spawn in a smartphone
3. Spawn in a spell book and scan it in
4. Spawn in a unidentified book and scan it in, marvel that the book you scanned is now identified.

#### Additional context
image

### CODE REVIEW

**Review by Maleclypse — CHANGES_REQUESTED** [review]
I'd prefer if this was jsonized so that mod creators can choose whether their "spell books" are capable of being added to an e-reader because otherwise I'm going to be receiving bug reports about "ball of pain was copied into my e-reader, how do you copy a 'ball of pain'?"

**Comment by Skrassh** [comment]
I'm ok with doing a Json addition then, I'll go and add a field for it.

**Comment by Skrassh** [comment]
Do we want to default the scannable field to true or false? If we want to default to false for books what's the best way to update the existing json aside from manually?

**Comment by MNG-cataclysm** [comment]
> Do we want to default the scannable field to true or false? If we want to default to false for books what's the best way to update the existing json aside from manually?

I think it should default to true and add a bool value to items that individuals don't want to be scannable. Something like `scannable: true/false` to item JSON that the game could check for.

**Comment by Skrassh** [comment]
> > Do we want to default the scannable field to true or false? If we want to default to false for books what's the best way to update the existing json aside from manually?
> 
> I think it should default to true and add a bool value to items that individuals don't want to be scannable. Something like `scannable: true/false` to item JSON that the game could check for.

That's what I was thinking, I'll go ahead and default that to true and get this wrapped up then.

**Comment by Maleclypse** [comment]
> > > Do we want to default the scannable field to true or false? If we want to default to false for books what's the best way to update the existing json aside from manually?
> > 
> > 
> > I think it should default to true and add a bool value to items that individuals don't want to be scannable. Something like `scannable: true/false` to item JSON that the game could check for.
> 
> That's what I was thinking, I'll go ahead and default that to true and get this wrapped up then.

As an in repo example please add `"scannable": false` to the Xedra Evolved spellbooks.  That way there's at least somewhere beyond docs it's visible.

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -2443,6 +2443,9 @@ void ebooksave_activity_actor::finish( player_activity &act, Character &who )
     ereader->put_in( book_copy, item_pocket::pocket_type::EBOOK );
     if( who.is_avatar() ) {
         add_msg( m_info, _( "You scan the book into your device." ) );
+        if( !who.has_identified( book->typeId() ) ) {
+            who.identify( *book );
+        }
```
[the remark this answers is not in the document]
Comment (Skrassh) [reply src/activity_actor.cpp:2448]: I moved it to before the scan 👍

**Comment by Skrassh** [comment]
> > > > Do we want to default the scannable field to true or false? If we want to default to false for books what's the best way to update the existing json aside from manually?
> > > 
> > > 
> > > I think it should default to true and add a bool value to items that individuals don't want to be scannable. Something like `scannable: true/false` to item JSON that the game could check for.
> > 
> > 
> > That's what I was thinking, I'll go ahead and default that to true and get this wrapped up then.
> 
> As an in repo example please add `"scannable": false` to the Xedra Evolved spellbooks. That way there's at least somewhere beyond docs it's visible.

I added scannable false to all the Xedra Evolved spell learning items.

**Review by Maleclypse — APPROVED** [review]
Thank you! This is a good enhancement!

**Comment by Skrassh** [comment]
Noticed when playing earlier that spellbooks report 0 mins to read on the identify check. Should we set a value for the spellbooks time to read or just not print the time if it's 0?

**Pull request merged**